condor_watch_q.py

```diff
@@ -420,6 +417,34 @@ def cli():
 GROUPBY_AD_KEY_TO_ATTRIBUTE = {v: k for k, v in GROUPBY_ATTRIBUTE_TO_AD_KEY.items()}
 
 
+def group_clusters(clusters, key, dagman_clusters_to_paths, batch_names, check_dagman):
+    groups = collections.defaultdict(list)
+    getter = operator.attrgetter(GROUPBY_AD_KEY_TO_ATTRIBUTE[key])
+    dagman_path_to_clusters = {v: k for k, v in dagman_clusters_to_paths.items()}
+
+    for cluster in clusters:
+        cluster_id = cluster.cluster_id
+        cluster_path = cluster.event_log_path
+
+        if cluster_path in dagman_path_to_clusters:
+            if key == BATCH_NAME:
+                dag_name = dagman_path_to_clusters[cluster_path]
+                groups[batch_names[dag_name]].append(cluster)
+            elif key == EVENT_LOG:
+                groups[cluster_path].append(cluster)
+            else:
+                groups[cluster_id].append(cluster)
```

Comment:
I don't think you need separate branches for these - shouldn't the same `getter` used for the non-DAG case work in these cases as well? I think the only special case is when sorting by batch name and also the job belongs to a DAG.